tests/unit/test_task_behavior.py

```diff
@@ -30,3 +30,53 @@ def test_redis_task_double_lookup(mock_redis):
     second_task = RedisTask(mock_redis, task_id)
     assert second_task.status == TaskState.SUCCESS
     assert second_task.internal_status == InternalTaskState.COMPLETE
+
+
+def test_redis_task_delete(mock_redis):
+    task_id = str(uuid.uuid1())
+
+    assert not RedisTask.exists(mock_redis, task_id)
+    task = RedisTask(mock_redis, task_id)
+
+    assert RedisTask.exists(mock_redis, task_id)
+    task.delete()
+
+    assert not RedisTask.exists(mock_redis, task_id)
+
+
+def test_task_group_create(mock_redis):
+    task_group_id = str(uuid.uuid1())
+    user_id = 101
+
+    assert not TaskGroup.exists(mock_redis, task_group_id)
+
+    tg = TaskGroup(mock_redis, task_group_id, user_id=user_id)
+    assert tg.user_id == user_id
+
+    assert TaskGroup.exists(mock_redis, task_group_id)
+
+
+def test_task_group_delete(mock_redis):
+    task_group_id = str(uuid.uuid1())
+    user_id = 101
+
+    assert not TaskGroup.exists(mock_redis, task_group_id)
+    tg = TaskGroup(mock_redis, task_group_id, user_id=user_id)
+    assert TaskGroup.exists(mock_redis, task_group_id)
+    tg.delete()
+    assert not TaskGroup.exists(mock_redis, task_group_id)
+
+
+def test_task_group_no_user_id(mock_redis):
+    # creating a TaskGroup without setting user_id results in no creation within redis,
+    # and the TaskGroup does not exist
+    # TODO: determine if this is the desired behavior
+    # maybe __init__ should raise an exception if `user_id` is not set?
```

Comment:
This is another oddity around the behavior that initialization and retrieval are conflated in the `RedisField` model.  To load a task group, you don't know a-priori the `user_id`, so you need it to be unspecified so it will load when accessed (and not be overwritten).  However, all newly created task groups *should* have a user_id.  Eventually it would probably be good to separate out load and create and have an explicit `load` or `save` method.

Reply:
This makes sense to me as a way of solving this short term, without the larger change of maybe getting rid of RedisField.

It would also allow for relatively simple enforcement of required fields.